fabric-level-generators-v1/src/main/java/net/fabricmc/fabric/mixin/level/generator/MixinPlayerRespawnS2CPacket.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.level.generator;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.client.network.packet.PlayerRespawnS2CPacket;
+import net.minecraft.world.level.LevelGeneratorType;
+
+@Mixin(PlayerRespawnS2CPacket.class)
+public class MixinPlayerRespawnS2CPacket {
```

Comment:
I'm not sure if this is against fabric's style, but this mixin is identical to the other packet one, and you can merge them into one mixin using `@Mixin({GameJoinS2CPacket.class, PlayerRespawnS2CPacket.class})`.

Also, all my comments on the other mixin apply to this mixin.

Reply:
@Dinolek why do you mark this as resolved?

Reply:
Mixin target is now different, because FIELD is targeted.
It can be moved to static method to limit duplicates.

Reply:
https://github.com/FabricMC/fabric/pull/483#discussion_r386059214

Reply:
In fact, the best way to do is to use networking login query to detect if the client knows this level generator type and accordingly decides whether to send the original or the vanilla default.

Reply:
https://github.com/FabricMC/fabric/pull/483#discussion_r386058379

Reply:
@liach Can we query that from vanilla client?

Reply:
Yes, we can use login query packet, and if client responds that it doesn't understand we can assume client doesn't have that level generator type and we should send vanilla's overworld level gen type as fallback

Reply:
Is there any example of this?

Reply:
Unfortunately no because adding login query support is a big job and fabric doesn't like big prs